tests/integration/test_schema_coordinator.py

```diff
@@ -82,12 +87,26 @@ async def get_client(
     await client.close()
 
 
+@retry(stop=stop_after_delay(RETRY_TIME), wait=wait_fixed(RETRY_WAIT_SECONDS))
+async def wait_for_ready(coordinator: SchemaCoordinator) -> None:
+    if not coordinator.ready and coordinator.coordinator_id is None:
+        raise TryAgain()
+
+
+@retry(stop=stop_after_delay(RETRY_TIME), wait=wait_fixed(RETRY_WAIT_SECONDS))
+async def wait_for_primary_state(coordinator: SchemaCoordinator) -> None:
```

Comment:
we should implement also an `async` version of the `retry`. Not sure if its smart enough to implement the `asyncio.wait` variant in case its decorating an async function.
In any case its an improvement for the flakiness that its the highest priority as now. Merging

Reply:
https://tenacity.readthedocs.io/en/latest/#async-and-retry

Jarkko provided me this doc, already solved.